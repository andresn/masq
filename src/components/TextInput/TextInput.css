.TextInput {
  display: flex;
  flex-direction: column-reverse;
  width: 100%;
}

.TextInput input {
  height: 33px;
  border-radius: 3px;
  border: solid 1px var(--grey2-color);
  font-size: 14px;
  padding-left: 8px;
}

.TextInput input:focus {
  outline: none;
  border: solid 1px var(--main-highlight-color);
}

.TextInput.error input:focus {
  border: solid 1px var(--error-color);
}

.TextInput input:focus + label {
  color: var(--main-highlight-color);
}

.TextInput.error input:focus + label {
  color: var(--error-color);
}

.TextInput label {
  margin: 0;
  font-size: 12px;
  margin-bottom: 8px;
  color: var(--grey3-color);
}
