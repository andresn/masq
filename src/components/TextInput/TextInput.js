import React from 'react'
import PropTypes from 'prop-types'

import './TextInput.css'

export default function TextInput (props) {
  const { error, label, labelError, defaultValue, onChange, password, focus } = props
  const classes = 'TextInput' + (error ? ' error' : '')
  const displayedLabel = (error && labelError) ? labelError : label

  return (
    <div className={classes}>
      <input type={password ? 'password' : 'text'} autoFocus={focus} defaultValue={defaultValue} onChange={onChange} onKeyUp={props.onKeyUp} />
      <label>{displayedLabel}</label>
    </div>
  )
}

TextInput.propTypes = {
  error: PropTypes.bool,
  label: PropTypes.string,
  onKeyUp: PropTypes.func,
  labelError: PropTypes.string,
  onChange: PropTypes.func,
  password: PropTypes.bool,
  defaultValue: PropTypes.string,
  focus: PropTypes.bool
}
