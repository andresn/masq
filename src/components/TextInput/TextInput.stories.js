import React from 'react'
import { storiesOf } from '@storybook/react'
import TextInput from './TextInput'

storiesOf('TextInput', module)
  .add('with one label', () => (
    <TextInput label='some input' />
  ))
  .add('with one label and a default value', () => (
    <TextInput label='some button' defaultValue='default text' />
  ))
  .add('with error', () => (
    <TextInput label='some button' error />
  ))
