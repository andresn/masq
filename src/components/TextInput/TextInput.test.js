import React from 'react'
import renderer from 'react-test-renderer'
import TextInput from './TextInput'

it('renders TextInput', () => {
  const component = renderer.create(
    <TextInput label='some button' />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders TextInput with a defaultValue', () => {
  const component = renderer.create(
    <TextInput label='some button' defaultValue='default text' />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders TextInput with error', () => {
  const component = renderer.create(
    <TextInput label='some button' error />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
