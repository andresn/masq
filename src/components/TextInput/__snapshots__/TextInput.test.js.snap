// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextInput 1`] = `
<div
  className="TextInput"
>
  <input
    defaultValue={undefined}
    onChange={undefined}
    type="text"
  />
  <label>
    some button
  </label>
</div>
`;

exports[`renders TextInput with a defaultValue 1`] = `
<div
  className="TextInput"
>
  <input
    defaultValue="default text"
    onChange={undefined}
    type="text"
  />
  <label>
    some button
  </label>
</div>
`;

exports[`renders TextInput with error 1`] = `
<div
  className="TextInput error"
>
  <input
    defaultValue={undefined}
    onChange={undefined}
    type="text"
  />
  <label>
    some button
  </label>
</div>
`;
